doc/source/f2py/buildtools/cmake.rst

```diff
@@ -0,0 +1,19 @@
+.. _f2py-cmake:
+
+===================
+Using via ``cmake``
+===================
+
+In terms of complexity, ``cmake`` falls between ``make`` and ``meson``. The
+learning curve is steeper since CMake syntax is not pythonic and is closer to
+``make`` with environment variables.
+
+However, the tradeoff is enhanced flexibility and support for most architectures
+and compilers. An introduction to the syntax is out of scope for this document,
```

Comment:
Also might be worth it to remark that CMake seems to be quite popular for building Fortran code (at least in my limited experience).

Reply:
TBH, I think `make` is still probably the most used when it comes to Fortran code, but I think mentioning that wouldn't be a good idea since `cmake` support is pretty poor for `f2py`, since it essentially consists of `add_custom_target` which is about as bad as it gets.

Reply:
`scikit-build` is the thing to recommend if you want to use CMake, it has f2py support and in general is "CMake for Python".

I'd personally just say "don't use `make`". It doesn't make sense on any platform in 2021, and users who still want that today choose their own pain and suffering.

Reply:
> scikit-build is the thing to recommend if you want to use CMake, it has f2py support and in general is "CMake for Python".

Thanks for the pointer, I'll check that out, too! I hope it fits my use case -- not a full-blown Python package (so no setup.py, nothing to "install"), just using python/pytest for testing/exercising an existing Fortran code. :-)

Also, nothing bars you from generating, e.g., Ninja instead of make files (but I'm sure y'all know that). I have to say, even on Windows make (via msys) seems to work well for me for C++/Fortran/both codebases. :-)

Reply:
Yup, but the `cmake` vanilla version is also slightly important for larger `cmake` based projects; but that's a good note; it shouldn't be used where it doesn't have to be.

`make` is definitely dead in the 21st century, but unfortunately there's a lot of legacy code with it ^_^